src/services/firebase.ts

```diff
@@ -112,7 +118,54 @@ export namespace ourFirebase {
     // Make sure we listen to match changes only once.
     // 'gamePortal/matches'
     // store.dispatch(updateMatchList);
-    db().ref('gamePortal/matches' + matchIds); // TODO
+    let tempMatchIds: Promise<MatchInfo>[] = [];
+    for (let matchId of matchIds) {
+      tempMatchIds.push(getMatchDetail(matchId));
+    }
+    Promise.all(tempMatchIds).then((matches: MatchInfo[]) => {
+      let action: Action = {
+        setMatchesList: matches
+      };
+      dispatch(action);
+    });
+
+    // db().ref('gamePortal/matches' + matchIds); // TODO
+  }
+
+  function getMatchDetail(matchId: string): Promise<MatchInfo> {
+    // let matchInfo = {};
+    return db()
+      .ref('gamePortal/matches/' + matchId)
+      .once('value')
+      .then((snap: firebase.database.DataSnapshot): MatchInfo => {
+        let matchFb: fbr.Match = snap.val();
+        let gameSpecId = matchFb.gameSpecId;
```

Comment:
Always be on safer side, reject when gameSpecId is null or snap.val() return null. Filter them on top.